File: app/models/foreman/hardware_model.rb
#
# Copyright 2013 Red Hat, Inc.
#
# This software is licensed to you under the GNU General Public
# License as published by the Free Software Foundation; either version
# 2 of the License (GPLv2) or (at your option) any later version.
# There is NO WARRANTY for this software, express or implied,
# including the implied warranties of MERCHANTABILITY,
# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
# have received a copy of GPLv2 along with this software; if not, see
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.

class Foreman::HardwareModel < Resources::ForemanModel
  include Resources::AbstractModel::IndexedModel

  attributes :name, :info, :vendor_class, :hardware_model
  validates :name, :presence => true

  resource_name :hardware_model
  foreman_resource_name :model

  def json_default_options
    return {
      :root => resource_name,
      :only => [:name, :info, :vendor_class, :hardware_model]
    }
  end

  index_options :display_attrs => [:name]

  mapping do
    indexes :id, :type=>'string', :index => :not_analyzed
    indexes :name, :type => 'string', :analyzer => :kt_name_analyzer
  end


end

File: cli/src/katello/client/api/hardware_model.py
# -*- coding: utf-8 -*-
#
# Copyright 2013 Red Hat, Inc.
#
# This software is licensed to you under the GNU General Public License,
# version 2 (GPLv2). There is NO WARRANTY for this software, express or
# implied, including the implied warranties of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
# along with this software; if not, see
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
#
# Red Hat trademarks are not licensed under GPLv2. No permission is
# granted to use or replicate Red Hat trademarks that are incorporated
# in this software or its documentation.

from urllib import quote

from katello.client.api.base import KatelloAPI
from katello.client.lib.utils.data import slice_dict


class HardwareModelAPI(KatelloAPI):

    @classmethod
    def __path(cls, model_id=None):
        if model_id is None:
            return "/api/hardware_models/"
        else:
            return "/api/hardware_models/%s/" % quote(str(model_id))

    def list(self):
        """
        list hardware model
        """
        return self.server.GET(self.__path())[1]


    def show(self, model_id):
        """
        show hardware model

        :type  model_id: string
        :param model_id: hardware model name
        """
        return self.server.GET(self.__path(model_id))[1]


    def create(self, data):
        """
        create hardware model

        :type  data['name']: string
        :param data['name']:
        :type  data['info']: string
        :param data['info']:
        :type  data['vendor_class']: string
        :param data['vendor_class']:
        :type  data['hardware_model']: string
        :param data['hardware_model']:
        """
        data = slice_dict(data, 'name', 'info', 'vendor_class', 'hardware_model')
        return self.server.POST(self.__path(), {"hardware_model": data})[1]


    def update(self, model_id, data):
        """
        update hardware model

        :type  model_id: string
        :param model_id: hardware model name
        :type  data: hash
        :param data: hardware model info
        :type  data['name']: string
        :param data['name']:
        :type  data['info']: string
        :param data['info']:
        :type  data['vendor_class']: string
        :param data['vendor_class']:
        :type  data['hardware_model']: string
        :param data['hardware_model']:
        """
        data = slice_dict(data, 'name', 'info', 'vendor_class', 'hardware_model')
        return self.server.PUT(self.__path(model_id), {"hardware_model": data})[1]


    def destroy(self, model_id):
        """
        destroy hardware model

        :type  model_id: string
        :param model_id: hardware model name
        """
        return self.server.DELETE(self.__path(model_id))[1]

